apps/web/src/components/Battle/BattleSettingsMenu.tsx

```
@@ -0,0 +1,107 @@
+  const settingRef = useRef<HTMLDivElement>(null);
+
+  const handleToggleSetting = useCallback(() => {
+    setIsSettingOpen((prev) => !prev);
+  }, []);
+
+  return (
+    <div className="relative" ref={settingRef}>
+      <button
+        onClick={handleToggleSetting}
+        className="rounded-full bg-base-faint p-2 text-ink shadow-sm transition hover:scale-110 active:scale-95"
+      >
```

Biome (2.3.13)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)